### DISCUSSION TITLE
Can Userprovider be optimized?

### DISCUSSION THREAD

**Discussion by wsy998** [laravel/framework#38231] [Ideas] [answered]
We don't need to "select * from table_ name " every time, especially Auth:: id(), which will greatly affect the database performance. We can add an attribute to the model to get what attributes we need.Besides, can I use model association after Auth:: user()?

**Comment by DarkGhostHunter** [comment, marked as answer]
Currently there is no way. The developer expects a `first()` equivalent call when dealing with the authenticated user.

You "could" use a Global Scope on the User model, which would be replicated on the `UserProvider`, but it would affect every query.

The only alternative is to extend the `EloquentUserProvider` with your own version, and override the `newModelQuery()` method, where the query gets created from a model class name.

```php
    /**
     * Get a new query builder for the model instance.
     *
     * @param  \Illuminate\Database\Eloquent\Model|null  $model
     * @return \Illuminate\Database\Eloquent\Builder
     */
    protected function newModelQuery($model = null)
    {
        $query = is_null($model)
                ? $this->createModel()->newQuery()
                : $model->newQuery();

        return $query->select('name', 'email');
    }
```

**Reply by wsy998** [reply]
Thank you for your suggestion. I'll try it
